mne/preprocessing/realign.py

```diff
@@ -85,25 +91,41 @@ def realign_raw(raw, other, t_raw, t_other, verbose=None):
         f"{raw.times[-1] * dr_ms_s:0.1f} ms)"
     )
 
-    # 2. Crop start of recordings to match using the zero-order term
-    msg = f"Cropping {zero_ord:0.3f} s from the start of "
+    # 2. Crop start of recordings to match
     if zero_ord > 0:  # need to crop start of raw to match other
-        logger.info(msg + "raw")
+        logger.info(f"Cropping {zero_ord:0.3f} s from the start of raw")
         raw.crop(zero_ord, None)
         t_raw -= zero_ord
     else:  # need to crop start of other to match raw
-        logger.info(msg + "other")
-        other.crop(-zero_ord, None)
-        t_other += zero_ord
+        logger.info(f"Cropping {(zero_ord / first_ord):0.3f} s from the start of other")
+        other.crop(-(zero_ord / first_ord), None)
+        t_other += zero_ord / first_ord
 
     # 3. Resample data using the first-order term
+    nan_ch_names = [
+        ch for ch in other.info["ch_names"] if np.isnan(other.get_data(picks=ch)).any()
+    ]
+    if len(nan_ch_names) > 0:  # Issue warning if any channel in other has nan values
+        warn(
+            f"Channel(s) {', '.join(nan_ch_names)} in other contain NaN values\n"
+            + "Resampling these channels will result in empty data\n"
+            + "(If realigning eye-tracking data, "
+            + "consider using interpolate_blinks and passing interpolate_gaze=True)"
```

Comment:
```suggestion
            f"Channel(s) {', '.join(nan_ch_names)} in `other` contain NaN values. "
            "Resampling these channels will result in the whole channel being NaN. "
            "(If realigning eye-tracking data, consider using interpolate_blinks and "
            "passing interpolate_gaze=True)"
```
You don't need `+` to concatenate strings across lines if they're within `(...)`.  Also we don't put newlines in most of our messages.